staging/src/k8s.io/client-go/transport/websocket/roundtripper.go

```diff
@@ -89,10 +90,8 @@ func (rt *RoundTripper) RoundTrip(request *http.Request) (retResp *http.Response
 	}
 	wsConn, resp, err := dialer.DialContext(request.Context(), request.URL.String(), request.Header)
 	if err != nil {
-		if err != gwebsocket.ErrBadHandshake {
-			return nil, err
-		}
-		return nil, fmt.Errorf("unable to upgrade connection: %v", err)
```

Comment:
is the switch to UpgradeFailureError to make the underlying error detectable? would `fmt.Errorf("unable to upgrade connection: %w", err)` work as well?

Reply:
This would work either way. I think it is slightly better documenting in the `FallbackExecutor` to check for the `UpgradeFailureError`.